### ISSUE TITLE
Bouncing Mechanics are frequently off

### ISSUE THREAD

**Issue by DimensionWarped** [Techokami/SonicWorldsNext#24]
Big example is if I do a big drop and hit an enemy, I sometimes just kill it and then immediately land. I'll be investigating this behavior soon.

**Comment by DimensionWarped** [comment]
Clearing assignment from myself as I haven't had much time to actively contribute lately. Hoping to resume working on new gimmicks soon though.

**Comment by TraceCalcum** [comment]
I fixed this problem in my own project by modifying the code in EnemyBase.gd by changing this code at line 19:
`if (i.movement.y < 0 or i.global_position.y-(i.velocity.y*delta) > global_position.y):
`
to:
`if (i.movement.y < 0):`

The second test in the if statement seems to return true in inappropriate circumstances, which causes the problem of the player sometimes simply landing after destroying an enemy. It also just seems unnecessary to me; is there ever a situation where you wouldn't want the player to bounce if they had downwards velocity? I cut this second test out and bouncing works perfectly.

This is just for enemies; monitors have their own problems, and the way I fixed them in my project is a little hacky, but I think it works fine. If you like, I can make a PR for this.
